pkg/apis/eksctl.io/v1alpha5/defaults.go

```diff
@@ -67,7 +67,7 @@ func SetNodeGroupDefaults(ng *NodeGroup, meta *ClusterMeta) {
 		ng.SSH.PublicKeyPath,
 		ng.SSH.PublicKey)
 
-	if numSSHFlagsEnabled > 0 {
+	if numSSHFlagsEnabled > 0 && ng.SSH.Allow == nil {
```

Comment:
What happens in this block if the config file has:

```
ssh:
  allow: true
  publicKeyPath: ./id_rsa_dev
```
or
```
ssh:
  allow: true
  keyName: ec2-dev-key
```
?

I think it would go to line 74 and override the key path.That's why I think it should only check for it being explicitly disabled.

If you want, you can simplify the logic a little with:

```
	if numSSHFlagsEnabled == 0 {
		if IsEnabled(ng.SSH.Allow) {
			ng.SSH.PublicKeyPath = &DefaultNodeSSHPublicKeyPath
		} else {
			ng.SSH.Allow = Disabled()
		}
	} else if !IsDisabled(ng.SSH.Allow) {
		// Enable SSH if not explicitly disabled when passing an SSH key
		ng.SSH.Allow = Enabled()
	}
```

Reply:
Ah nice find! I've applied your changes and ran through the tests again, everything is working as expected.

Thanks for your patience, I haven't done a lot of Golang programming.

Reply:
Thank you  for the contribution! We appreciate it!